sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ComponentHelper.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Exporter.AzureMonitor
+{
+    internal class ComponentHelper
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void ExtractComponentProperties(Dictionary<string, string> tags, ActivityKind activityKind, out string type, out string sourceOrTarget)
+        {
+            sourceOrTarget = null;
+            type = null;
+
+            if (!tags.TryGetValue(SemanticConventions.AttributeAzureNameSpace, out var component))
+            {
+                tags.TryGetValue(SemanticConventions.AttributeComponent, out component);
+            }
+
+            if (component == null)
+            {
+                return;
+            }
+
+            string kind = null;
+
+            switch (activityKind)
+            {
+                case ActivityKind.Internal:
+                    kind = "InProc";
+                    break;
+                case ActivityKind.Producer:
+                    kind = "Queue Message";
+                    break;
+                default:
+                    break;
+            }
+
+            switch (component)
+            {
+                case "eventhubs":
+                case "Microsoft.EventHub":
+                    type = kind == null ? RemoteDependencyConstants.AzureEventHubs : string.Concat(kind, " | ", RemoteDependencyConstants.AzureEventHubs);
```

Comment:
This looks excessive per call, there's only 2 outputs of this path where kind != null and they are constants.

Reply:
Fixed.